.github/actions/sanitize/action.yml

```diff
@@ -40,23 +54,152 @@ runs:
 
     # https://github.com/actions/setup-node/issues/899
     - name: Enable Corepack before setting up Node
+      if: ${{ inputs.test-grafbase-gateway == 'false' }}
       shell: bash
       run: corepack enable
 
     - name: Setup Node.js
+      if: ${{ inputs.test-grafbase-gateway == 'false' }}
       uses: actions/setup-node@v4
       with:
         node-version: 18
 
     - name: Install pnpm
+      if: ${{ inputs.test-grafbase-gateway == 'false' }}
       uses: pnpm/action-setup@v2
       id: pnpm-install
       with:
         version: 8
         run_install: false
 
-    - if: ${{ inputs.test == 'true' }}
-      name: cargo nextest
+    - name: Login to Docker Hub
+      if: ${{ inputs.test-engine == 'true' }}
+      uses: docker/login-action@v3
+      with:
+        username: ${{ inputs.dockerhub-username }}
+        password: ${{ inputs.dockerhub-token }}
+
+    - name: Start local databases
+      shell: bash
+      if: ${{ inputs.test-engine == 'true' }}
+      run: |
+        docker-compose -f engine/crates/integration-tests/docker-compose.yml up -d
+
+    - if: ${{ inputs.test-engine == 'true' }}
```

Comment:
If everything is behind an `if`, I would suggest creating a separate workflow that you would call like:
https://github.com/grafbase/api/blob/main/.github/workflows/deploy.yml#L289

it would be a lot easier to manage and understand imo